### PR TITLE
Add k6 testing

### PR DESCRIPTION
**Pull request by alyssabull**
This PR adds k6 e2e testing.

### CODE REVIEW

File: e2e/e2ek6.test.js [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+    // checks the page for the data source is working message
+    check(page, {
+      'add datasource successful':
+      await page.locator('[aria-label="Create a dashboard').textContent() === "building a dashboard",
+    });
+
+    const orgName = `api-org-${uuidv4()}`
+
+    // creates org
+    const getOrg = http.post('http://admin:admin@localhost:3000/api/orgs', `{"name": "${orgName}"}`, {
```
Comment (aangelisc) [inline e2e/e2ek6.test.js:71]: Why do these steps need to be carried out if the save and test flow succeeds? Are you effectively adding the datasource twice?

File: e2e/e2ek6.test.js [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+    
+    const saveDashboardToolbarButton = page.locator(`button[aria-label="${selectors.components.PageToolbar.item('Save dashboard')}"]`);
+    await saveDashboardToolbarButton.click();
+    const dashboardTitleInput = page.locator(`input[aria-label="${selectors.pages.SaveDashboardAsModal.newName}"]`);
+    dashboardTitleInput.fill('');
+    await dashboardTitleInput.type(DASHBOARD_TITLE);
+    const saveDashboardModalButton = page.locator(`button[aria-label="${selectors.pages.SaveDashboardAsModal.save}"]`);
+    saveDashboardModalButton.click();
+
+    // checks that query is run successfully
```
Comment (aangelisc) [inline e2e/e2ek6.test.js:141]: nit: comment just needs updating

File: e2e/e2ek6.test.js [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+              "datasourceId": 2922,
+              "intervalMs": 30000,
+              "maxDataPoints": 638
+          }
+      ],
+      "from": "1686147681613",
+      "to": "1686169281613"
+    };
+
+    const res = http.post('http://localhost:3000/api/ds/query/', JSON.stringify(queryData), {
```
Comment (aangelisc) [inline e2e/e2ek6.test.js:259]: Same question as above for this

File: e2e/e2ek6.test.js [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+    check(res, {
+      'run query returns a response of 200':
+      (r) => r.status === 200
+    })
+  } catch(e) {
+    fail(`run query failed: ${e}`);
+  } 
+};
+
+export async function removeDatasource(browser, page) {
```
Comment (aangelisc) [inline e2e/e2ek6.test.js:273]: I think this should be `removeDashboard`

**Review by aangelisc — APPROVED** [review]
Great work on this @alyssabull! Thank you for dogfooding k6 and adding some much needed E2E tests. Some nits and questions in the comments but otherwise this looks great to me 😊

**Review by asimpson — APPROVED** [review]
Looks great, so excited for this! I have the same nits as @aangelisc btw :)

File: e2e/e2ek6.test.js (hunk 1 above)
Comment (alyssabull) [reply e2e/e2ek6.test.js:71]: You are correct re:adding the datasource twice - good catch. The idea was to check the status code of `checkHealth` not just the success message. Also by creating a datasource this way it was giving access to the `datasouceUID` which was allowing a query to be ran in a later test. I think what makes most sense now is to remove this flow and keep the first one (typing in localhost/9000, etc.) I will amend the `configurePanel` test to get the UID another way.

**Review by aangelisc — DISMISSED** [review]
This LGTM, let's get it in! Nice work @alyssabull 😊

File: e2e/e2ek6.test.js (hunk 3 above)
Comment (alyssabull) [reply e2e/e2ek6.test.js:259]: I am going to match the [grafana `configurePanel` test](https://github.com/grafana/grafana/blob/d6c468c1c24fa70e2a0cfab6e5f8a98ed259e13d/packages/grafana-e2e/src/flows/configurePanel.ts#L112) and check the response of the POST request here.

File: e2e/e2ek6.test.js [hunk 5]
```diff
@@ -56,7 +55,8 @@ export async function addDatasource(page) {
     serverAddress.type('localhost');
     const serverPort = page.locator('input[aria-label="Server port"]');
     serverPort.type('9000');
-    const saveAndTestButton = page.locator(`button[data-testid="data-testid ${selectors.pages.DataSource.saveAndTest}"]`);
+    // const saveAndTestButton = page.locator(`button[data-testid="data-testid ${selectors.pages.DataSource.saveAndTest}"]`);
```
Comment (aangelisc) [inline e2e/e2ek6.test.js:58]: I'm assuming there are problems with this selector?

**Review by aangelisc — APPROVED** [review]

File: e2e/e2ek6.test.js [hunk 6]
```diff
@@ -194,16 +193,20 @@ export async function configurePanel(page) {
...
       headers: { 'Content-Type': 'application/json' },
     });
     apiToken = getApiToken.json().key;
+    console.log('apiToken', getApiToken.json())
 
     // sends POST request for query
     const res = http.post('http://admin:admin@localhost:3000/api/ds/query/', JSON.stringify(queryData), {
       headers: { 'Content-Type': 'application/json', 'Authorization': `Bearer ${apiToken}` },
     });
+    console.log('res', res)
```
Comment (aangelisc) [inline e2e/e2ek6.test.js:209]: Just remove the excess logs 😊

File: e2e/e2ek6.test.js [hunk 7]
```diff
@@ -219,6 +222,7 @@ export async function removeDashboard(browser, page) {
   try {
     const dashboardURL = page.url();
     const dashboardUID = getDashboardUid(dashboardURL);
+    console.log('dashboardUID', dashboardUID)
```
Comment (aangelisc) [inline e2e/e2ek6.test.js:225]: And this one

**Review by aangelisc — DISMISSED** [review]

File: e2e/e2ek6.test.js (hunk 6 above)
Comment (asimpson) [reply e2e/e2ek6.test.js:209]: Same :+1:

**Review by asimpson — APPROVED** [review]
Same comment as @aangelisc, remove the `console.log` statements and this is good to go :muscle: nice work!

**Review by asimpson — APPROVED** [review]
It's so good :trophy: really good work!

**Review by aangelisc — APPROVED** [review]
Let's get this merged 😄

**Pull request merged**